docs/inventory_plugins.md

```diff
@@ -0,0 +1,197 @@
+eliminates the need to maintain source-specific logic,
+relying on Ansible's code instead. This also allows us to
+count on a consistent data structure outputted from `ansible-inventory`.
+There are many valid structures that a script can provide, but the output
+from `ansible-inventory` will always be the same,
+thus the AWX logic to parse the content is simplified.
+This is why even scripts must be ran through the `ansible-inventory` CLI.
+
+Along with this switchover, a backported version of
+`ansible-inventory` was provided that supported Ansible versions 2.2 and 2.3.
+
+### Removal of Backport
+
+In AWX 3.0.0 (and Tower 3.5), the backport of `ansible-inventory`
+was removed, and support for using custom virtual environments was added.
+This set the minimum version of Ansible necessary to run _any_
+inventory update to 2.4.
+
+## Inventory Plugin Versioning
+
+Beginning in Ansible 2.5, cloud sources in Ansible started migrating
+away from "contrib" scripts (meaning they lived in the contrib folder)
+to the inventory plugin model.
+
+In AWX 4.0.0 (and Tower 3.5) inventory source types start to switchover
+to plugins, provided that sufficient compatibility is in place for
+the version of Ansible present in the custom virtualenv where the import
+is being ran.
+
+To see what version the plugin transition will happen, see
+`awx/main/models/inventory.py` and look for the source name as a
+subclass of `PluginFileInjector`, and there should be an `initial_version`
+which is the first version that testing deemed to have sufficient parity
+in the content its inventory plugin returns. For example, `openstack` will
+begin using the inventory plugin in Ansible version 2.8, because the
+openstack inventory plugin had an issue with sending logs to stdout which
+was fixed in that version. If you run an openstack inventory update in
+2.8 or lower, it will use the script.
+
+### Sunsetting the scripts
+
+Eventually, it is intended that all source types will have moved to
+plugins. For any given source, after the `initial_version` for plugin use
+is higher than the lowest supported Ansible version, the script can be
+removed and the logic for script credential injection can also be removed.
+
+For example, after AWX no longer supports Ansible 2.7, the script
+`awx/plugins/openstack_inventory.py` will be removed.
+
+## Changes to Expect in Imports
+
+An effort was made to keep imports working in the exact same way after
+the switchover. However, the inventory plugins are a fundamental rewrite
+and many elements of default behavior has changed. Because of that,
+a `compatibility_mode` toggle was added.
+
+In a data migration, all existing cloud sources are switched over to
+use `compatibility_mode`. New inventory sources will default to having
+this off.
+
+We recommend that you opt out of compatibility mode because this is more
+future-proof, and also suggest that you set the `overwrite`
+flag to help assure stale content is removed.
```

Comment:
I would be stronger here. If you flip the compatibility mode setting, you really need to set `overwrite` and `overwrite_vars`.